packages/accounts-password/password_client.js

```diff
@@ -116,7 +116,7 @@ Accounts.createUser = function (options, callback) {
   if (typeof options.password !== 'string')
     throw new Error("options.password must be a string");
```

Comment:
I think type checking errors need to be thrown as exceptions to get noticed during development, so the developer will realise they're sending in the wrong thing when they test it.
On the other hand, unhandled exceptions in production are ugly for the users.